routing/transit_graph.cpp

```diff
@@ -0,0 +1,231 @@
+    return RouteWeight(GetGate(segment).GetWeight(), 0 /* nontransitCross */);
+
+  if (IsEdge(segment))
+    return RouteWeight(GetEdge(segment).GetWeight(), 0 /* nontransitCross */);
+
+  // Estimate weight for part of real and projections.
+  return RouteWeight(estimator.CalcLeapWeight(GetJunction(segment, false /* front */).GetPoint(),
+                                              GetJunction(segment, true /* front */).GetPoint()),
+                     0 /* nontransitCross */);
+}
+
+void TransitGraph::GetTransitEdges(Segment const & segment, bool isOutgoing,
+                                   vector<SegmentEdge> & edges,
+                                   EdgeEstimator const & estimator) const
+{
+  CHECK(IsTransitSegment(segment), ("Nontransit segment passed to TransitGraph."));
+  for (auto const & s : m_fake.GetEdges(segment, isOutgoing))
+    edges.emplace_back(s, CalcSegmentWeight(isOutgoing ? s : segment, estimator));
+}
+
+set<Segment> const & TransitGraph::GetFake(Segment const & real) const
+{
+  return m_fake.GetFake(real);
+}
+
+bool TransitGraph::FindReal(Segment const & fake, Segment & real) const
+{
+  return m_fake.FindReal(fake, real);
+}
+
+void TransitGraph::Fill(vector<transit::Stop> const & stops, vector<transit::Gate> const & gates,
+                        vector<transit::Edge> const & edges, EdgeEstimator const & estimator,
+                        NumMwmId numMwmId, IndexGraph & indexGraph)
+{
+  m_mwmId = numMwmId;
+  map<transit::StopId, Junction> stopCoords;
+  for (auto const & stop : stops)
+    stopCoords[stop.GetId()] = Junction(stop.GetPoint(), feature::kDefaultAltitudeMeters);
+
+  for (auto const & gate : gates)
+  {
+    // TODO (@t.yan) after https://github.com/mapsme/omim/pull/7240 merge
+    // auto const gateSegment = gate.GetBestPedestrianSegment();
+    // Segment real(numMwmId, gateSegment.GetFeatureId(), gateSegment.GetSegmentIdx(), gateSegment.GetForward());
+    auto const ending = MakeFakeEnding( Segment() /* real */, gate.GetPoint(), estimator, indexGraph);
+    if (gate.GetEntrance())
+      AddGate(gate, ending, stopCoords, true /* isEnter */);
+    if (gate.GetExit())
+      AddGate(gate, ending, stopCoords, false /* isEnter */);
+  }
+
+  map<transit::StopId, set<transit::Edge>> outgoing;
+  map<transit::StopId, set<transit::Edge>> ingoing;
+  for (auto const & edge : edges)
+  {
+    AddEdge(edge, stopCoords);
+    outgoing[edge.GetStartStopId()].insert(edge);
+    ingoing[edge.GetFinishStopId()].insert(edge);
+  }
+
+  AddConnections(outgoing, true /* isOutgoing */);
+  AddConnections(ingoing, false /* isOutgoing */);
+}
+
+Segment TransitGraph::GetTransitSegment(uint32_t segmentIdx) const
+{
+  return Segment(m_mwmId, kTransitFeatureId, segmentIdx, false);
+}
+
+Segment TransitGraph::GetNewTransitSegment() const
+{
+  CHECK_LESS_OR_EQUAL(m_fake.GetSize(), std::numeric_limits<uint32_t>::max(), ());
+  return GetTransitSegment(static_cast<uint32_t>(m_fake.GetSize()));
```

Comment:
Please, add inline comment to the param.

Reply:
done